src/scripts/modules/media/nav/nav.coffee

```diff
@@ -147,28 +148,31 @@ define (require) ->
       removeClass('fa-search').
       addClass('fa-spinner fa-spin load-search')
 
+    handleSearch: ->
+      @searchTerm = @$el.find('.searchbar input').val()
+      if @searchTerm == ''
+        @clearSearch()
+        return
+      options = {
+        bookId: "#{@model.get('id')}@#{@model.get('version')}",
+        query: @searchTerm
+      }
+      # before the search has loaded
+      @enableLoadSearch()
+      # after the search has completed
+      BookSearchResults.fetch(options).done((data) =>
+        if not @tocIsOpen
+          @toggleContents()
+        @model.set('searchResults', data.results)
+        @enableClearSearch()
+      ).fail((err) ->
+        console.error("Search failed:", err)
```

Comment:
ooh, is there a quick way to let the user know the search failed? It's not a regression since the original code silently failed as well.